src/shogun/regression/KernelRidgeRegression.cpp

```diff
@@ -43,82 +42,44 @@ void CKernelRidgeRegression::init()
 	SG_ADD(&m_tau, "tau", "Regularization parameter", MS_AVAILABLE);
 }
 
-bool CKernelRidgeRegression::train_machine_pinv()
+bool CKernelRidgeRegression::solve_krr_system()
 {
-	// Get kernel matrix
-	SGMatrix<float64_t> kernel_matrix=kernel->get_kernel_matrix<float64_t>();
-	int32_t n = kernel_matrix.num_cols;
-	int32_t m = kernel_matrix.num_rows;
-	ASSERT(kernel_matrix.matrix && m>0 && n>0)
-
-	for(int32_t i=0; i < n; i++)
-		kernel_matrix.matrix[i+i*n]+=m_tau;
-
-	/* re-set alphas of kernel machine */
-	m_alpha=((CRegressionLabels*) m_labels)->get_labels_copy();
-
-	/* tell kernel machine that all alphas are needed as'support vectors' */
-	m_svs=SGVector<index_t>(m_alpha.vlen);
-	m_svs.range_fill();
-
-	if (get_alphas().vlen!=n)
+	SGMatrix<float64_t> kernel_matrix(kernel->get_kernel_matrix());
+	if (kernel_matrix.num_rows!=kernel_matrix.num_cols)
+	{
```

Comment:
I think this check should not be necessary. In fact, the algorithm (or the base class have to check) should make sure that the kernel is initialised with the training data on both sides.....